lib/node_modules/@stdlib/array/base/map/lib/main.js

```diff
@@ -0,0 +1,161 @@
+* @param {string} method - method name
+* @returns {boolean} boolean indicating whether an object has a specified method
+*
+* @example
+* var bool = hasMethod( [], 'map' );
+* // returns true
+*
+* @example
+* var bool = hasMethod( [], 'beep' );
+* // returns false
+*/
+function hasMethod( obj, method ) {
+	return ( typeof obj[ method ] === 'function' );
+}
+
+/**
+* Applies a callback function to elements in an input array and assigns results to elements in a new output array.
+*
+* @private
+* @param {Collection} x - input array
+* @param {Function} fcn - callback function
+* @param {*} thisArg - callback execution context
+* @returns {Collection} output array
+*
+* @example
+* var ones = require( '@stdlib/array/base/ones' );
+*
+* function scale( x ) {
+*     return x * 10.0;
+* }
+*
+* var x = ones( 4 );
+* var y = internal( x, scale );
+* // returns [ 10.0, 10.0, 10.0, 10.0 ]
+*/
+function internal( x, fcn, thisArg ) {
+	var out;
+	var i;
+
+	out = zeros( x.length );
+	for ( i = 0; i < x.length; i++ ) {
+		out[ i ] = fcn.call( thisArg, x[ i ], i, x );
+	}
+	return out;
+}
+
+/**
+* Applies a callback function to elements in an input array and assigns results to elements in a new output array.
+*
+* @private
+* @param {Object} x - input array object
+* @param {Function} fcn - callback function
+* @param {*} thisArg - callback execution context
+* @returns {Object} output array object
+*
+* @example
+* var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
+* var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+* var ones = require( '@stdlib/array/base/ones' );
+*
+* function scale( x ) {
+*     return x * 10.0;
+* }
+*
+* var x = arraylike2object( toAccessorArray( ones( 4 ) ) );
+* var y = accessors( x, scale );
+* // returns {...}
+*/
+function accessors( x, fcn, thisArg ) {
+	var xdata;
+	var odata;
+	var xget;
+	var oset;
+	var o;
+	var i;
+
+	xdata = x.data;
+	xget = x.accessors[ 0 ];
+
+	o = arraylike2object( toAccessorArray( zeros( xdata.length ) ) );
+	odata = o.data;
+	oset = o.accessors[ 1 ];
+
+	for ( i = 0; i < xdata.length; i++ ) {
+		oset( odata, i, fcn.call( thisArg, xget( xdata, i ), i, xdata ) );
+	}
+	return o;
+}
+
+
+// MAIN //
+
+/**
+* Applies a callback function to elements in an input array and assigns results to elements in a new output array.
+*
+* @param {Collection} x - input array
+* @param {Function} fcn - callback function
+* @param {*} [thisArg] - callback execution context
+* @returns {Collection} output array
+*
+* @example
+* var ones = require( '@stdlib/array/base/ones' );
+*
+* function scale( x ) {
+*     return x * 10.0;
+* }
+*
+* var x = ones( 4 );
+* var y = map( x, scale );
+* // returns [ 10.0, 10.0, 10.0, 10.0 ]
+*/
+function map( x, fcn, thisArg ) {
+	var obj;
+	if ( hasMethod( x, 'map' ) ) {
+		return x.map( fcn, thisArg );
+	}
+	obj = arraylike2object( x );
+	if ( obj.accessorProtocol ) {
+		return accessors( obj, fcn, thisArg );
```

Comment:
This is incorrect. We should NOT be returning `obj`.

For both here and L155, you should have already preallocated a generic array at L151, and then you should pass that output array to `accessors` and `internal`, rather than allocating within those functions.